assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.js

```diff
@@ -0,0 +1,55 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { isBlank } from "@ember/utils";
+import { action, computed } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class MoveToChannelModalInner extends Component {
+  @service chat;
+  @service router;
+  tagName = "";
+  sourceChannel = null;
+  destinationChannelId = null;
+  selectedMessageIds = null;
+
+  @computed("destinationChannelId")
+  get disableMoveButton() {
+    return isBlank(this.destinationChannelId);
+  }
+
+  @discourseComputed("chat.publicChannels.[]")
+  availableChannels(publicChannels) {
+    return publicChannels.reject((chan) => chan.id === this.sourceChannel.id);
+  }
+
+  get selectedMessageCount() {
+    return this.selectedMessageIds.length;
+  }
+
+  @action
+  moveMessages() {
+    return ajax(
+      `/chat/${this.sourceChannel.id}/move_messages_to_channel.json`,
+      {
+        method: "PUT",
+        data: {
+          message_ids: this.selectedMessageIds,
+          destination_channel_id: this.destinationChannelId,
+        },
+      }
+    )
+      .then((response) => {
+        return this.router.transitionTo(
```

Comment:
I think you can return a promise from this with .promise I think

Reply:
Okay, but I don't think it really needs any return value since nothing is waiting on it. Will just remove return.